stdlib/asm/collections/mmr.masm

```diff
@@ -175,3 +196,47 @@ export.unpack
   # assert on the resulting hash (11 cycles)
   assert_eqw
 end
+
+#! Computes the hash of the given Mmr.
+#!
+#! Input: [mmr_ptr, ...]
+#! Output: [HASH, ...]
+#! Cycles: 123 + 3 * num_peaks
+export.pack
+  # load num_leaves (2 cycles)
+  dup mem_load
+  # => [num_leaves, mmr_ptr, ...]
+
+  # compute num_peaks (69 cycles)
+  exec.num_leaves_to_num_peaks
+  # => [num_peaks, mmr_ptr, ...]
+
+  # compute the message size (18 cycles)
+  exec.num_peaks_to_message_size
+  # => [message_size, mmr_ptr, ...]
+
+  # prepare the arguments for adv.mem
+  add.1 swap
+  # => [mmr_ptr, number_words, ...]
+
+  # compute peaks_start and peaks_end (5 cycles)
+  dup.1 dup.1 add dup.1 add.1
+  # => [peaks_start, peaks_end, mmr_ptr, number_words, ...]
+
+  # hash the memory contents (25 + 3 * num_peaks)
+  padw padw padw
+  exec.native::hash_even_memory_addresses
+  exec.native::digest_output
+  # => [HASH, peaks_end, peaks_end, peaks_start, num_leaves, ...]
```

Comment:
If `adv.mem` was defined with `start_address` and `end_address` instead of `number_of_words`, I could reuse `peaks_end` and maybe save a few cycles.

I haven't changed it because I'm not sure if that would be best in general.